src/common/init.cpp

```diff
@@ -200,17 +200,33 @@ void wxInitData::MSWInitialize()
     // argvMSW because it could be allocated by Initialize() if a custom entry
     // point is used.
     argv = argvMSW;
+
+#ifndef __WXMSW__
+    // We need to convert from wide arguments back to the narrow ones.
+    argvA = new char*[argc + 1];
+    argvA[argc] = nullptr;
+
+    ownsArgvA = true;
+
+    for ( int i = 0; i < argc; i++ )
+    {
+        // Try to use the current encoding, but if it fails, it's better to
+        // fall back to UTF-8 than lose an argument entirely.
+        argvA[i] = wxConvWhateverWorks.cWC2MB(argv[i]).release();
+    }
+#endif // !__WXMSW__
```

Comment:
Now the logic indeed makes sense to me, thanks!

But this code seems to exactly duplicate the code below, so what about extracting it into `InitArgsA()` and calling it from both places instead of copying it?

Reply:
> But this code seems to exactly duplicate the code below, so what about extracting it into `InitArgsA()` and calling it from both places instead of copying it?

Makes sense to me, will do!

Reply:
Would you prefer it be a helper function (taking in the values of argc and argv (well, the pointer at least) and a reference to argvA and building it) or a member function of wxInitData?

I'm not sure what the philosophy is here. Since the include is already in a `private` subdirectory, it seems fine to add helpers as members there?

Reply:
Yes, it's definitely fine to add a member function to do it, TIA!